draco/schema.py

```diff
@@ -1,15 +1,50 @@
 from math import e
 from pathlib import Path
-from typing import Any, Dict
+from typing import Any, List, Literal, TypedDict, Union
 
 import numpy as np
 import pandas as pd
 
+# Field types recognized by a Draco schema.
+FieldType = Literal["number", "string", "boolean", "datetime"]
 
-def dtype_to_field_type(ty):
-    """Simple converter that translates Pandas column types to data types for
-    Draco.
-    """
+
+class BaseFieldProps(TypedDict):
+    """Properties shared by fields of all types in a `Schema`."""
+
+    name: str
+    type: FieldType
+    unique: int
+    entropy: float
+
+
+class NumberFieldProps(BaseFieldProps):
```

Comment:
I wouldn't consider this very stable API, though. We may add new properties from time to time. Just a note and no todo.

Reply:
I also assumed that we will extend the schema as we go, but this PR introduces a primitive typing base which we will be able to extend whenever adding new properties.